app/models/fact_value.rb

```diff
@@ -81,13 +81,16 @@ def self.count_each(fact, options = {})
     output
   end
 
-  def self.build_facts_hash(facts)
-    hosts = Host.where(:id => facts.group_by(&:host_id).keys).all
-
+  def self.build_facts_hash(facts, has_host = false)
```

Comment:
I think we need more descriptive name than `has_host`. I would suggest `single_host`.